### PR TITLE
feat: add arbitrum

### PR DESCRIPTION
**Pull request by janndriessen**
## **Summary of Changes**

* Adds Arbitrum chain
* Introduces handling around multi chain e.g. making sure that only desired chains are connected for the individual trade UI's
* Adds all new index tokens (and updates Arbitrum token addresses)
* Adds approvals code
* Updates FlashMintSDK (for new dsETH issuance)

## **Test Data or Screenshots**

image

###### _By submitting this pull request, you are confirming the following to be true:_

- I have reviewed the [Contribution Guidelines](https://github.com/IndexCoop/index-app/blob/master/CONTRIBUTING.md).
- I have performed a self-review of my own code.
- I have updated my repository to match the master at `IndexCoop/index-app`.
- I have included test data or screenshots that prove my fix is effective or that my feature works.

### CODE REVIEW

File: src/app/leverage/components/leverage-widget/index.tsx [hunk 1]
```diff
@@ -49,6 +54,17 @@ export function LeverageWidget() {
   const { inputBalance, inputBalanceFormatted, isFetchingQuote } =
     useFormattedLeverageData(stats)
 
+  const {
+    isApproved,
+    isApproving,
+    approve: onApprove,
+  } = useApproval(
+    inputToken,
+    // FIXME: change to correct FlashMint contract
```
Comment (0xonramp) [inline src/app/leverage/components/leverage-widget/index.tsx:63]: 👍

File: src/constants/tokens.ts [hunk 2]
```diff
@@ -136,9 +141,29 @@ export const IndexCoopBitcoin2xIndex: Token = {
   url: 'btc2x',
 }
 
-const eth2x = getTokenFromSymbol('ETH2X')!
+const btc3xArbitrum = getIndexTokenData('BTC3X', ARBITRUM.chainId)!
+export const IndexCoopBitcoin3xIndex: Token = {
+  ...btc3xArbitrum,
+  address: '',
```
Comment (0xonramp) [inline src/constants/tokens.ts:147]: is this a TODO, or intend for the address to be an empty string?

File: src/constants/tokens.ts [hunk 3]
```diff
@@ -136,9 +141,29 @@ export const IndexCoopBitcoin2xIndex: Token = {
...
+  arbitrumAddress: btc3xArbitrum.address,
+  // Random for now - as no listing
+  coingeckoId: 'btc3x',
+  fees: {
+    streamingFee: '	3.65%',
```
Comment (0xonramp) [inline src/constants/tokens.ts:152]:
```suggestion
    streamingFee: '3.65%',
```

File: src/constants/tokens.ts (hunk 2 above)
Comment (janndriessen) [reply src/constants/tokens.ts:147]: Intended to be empty - as it's not available on mainnet 👍

File: src/app/leverage/components/leverage-widget/index.tsx [hunk 4]
```diff
@@ -65,11 +81,11 @@ export function LeverageWidget() {
     onClose: onCloseTransactionReview,
   } = useDisclosure()
 
-  const isApproved = true
-  const isApproving = false
+  // TODO:
```
[the remark this answers is not in the document]
Comment (janndriessen) [reply src/app/leverage/components/leverage-widget/index.tsx:84]: To add `hasInsufficientFunds`. I'll work on that tomorrow. 🤝

**Pull request merged**